src/openrct2/Game.cpp

```diff
@@ -530,11 +528,10 @@ void game_load_init()
     {
         viewport_init_all();
         game_create_windows();
-        mainWindow = window_get_main();
     }
     else
     {
-        mainWindow = window_get_main();
+        const auto mainWindow = window_get_main();
```

Comment:
Shouldn’t this be

```suggestion
        const auto* mainWindow = window_get_main();
```
?

Reply:
With and without explicit `*` seems to be a bit mixed.

```
src/openrct2-ui/WindowManager.cpp:527:        auto mainWindow = window_get_main();
src/openrct2-ui/input/InputManager.cpp:115:    auto mainWindow = window_get_main();
src/openrct2-ui/input/Shortcuts.cpp:60:        auto window = window_get_main();
src/openrct2-ui/input/Shortcuts.cpp:72:        auto window = window_get_main();
src/openrct2-ui/windows/RideConstruction.cpp:2141:            auto* mainWnd = window_get_main();
src/openrct2-ui/windows/TopToolbar.cpp:837:    const auto* mainWindow = window_get_main();
src/openrct2-ui/windows/TopToolbar.cpp:1217:                auto* mainWnd = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:87:        auto* mainWindow = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:106:        auto* mainWindow = window_get_main();
src/openrct2-ui/windows/Viewport.cpp:142:                auto* mainWindow = window_get_main();
src/openrct2/ReplayManager.cpp:862:                    auto* mainWindow = window_get_main();
src/openrct2/drawing/LightFX.cpp:305:                auto* w = window_get_main();
src/openrct2/interface/Screenshot.cpp:393:        auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1800:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1830:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1860:    auto mainWindow = window_get_main();
src/openrct2/interface/Window.cpp:1890:    auto mainWindow = window_get_main();
src/openrct2/interface/Window_internal.cpp:20:    auto mainWindow = window_get_main();
```

Reply:
I think there is a general movement towards always using `auto*` if the variable in question is a pointer in our codebase. (Would be lovely if we could enforce that automatically using clang-format or something like that.)

Reply:
Sounds like a good idea, that also helps finding instances that could be moved over to std::shared_ptr more easily.